transaction.go

```diff
@@ -0,0 +1,489 @@
+package factom
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"time"
+
+	"github.com/Factom-Asset-Tokens/factom/varintf"
+)
+
+type FactoidTransaction struct {
+	// TODO: The header is usually at the top level, is this ok?
+	FactoidTransactionHeader
+
+	FCTInputs  []FactoidTransactionIO        `json:"inputs"`
+	FCTOutputs []FactoidTransactionIO        `json:"fctoutputs"`
+	ECOutputs  []FactoidTransactionIO        `json:"ecoutputs"`
+	Signatures []FactoidTransactionSignature `json:"signatures"`
+}
+
+type FactoidTransactionHeader struct {
+	// TransactionID is not in the marshalled binary
+	TransactionID *Bytes32 `json:"txid"`
+
+	Version uint64 `json:"version"`
+	// Timestamp is accurate to the millisecond
+	Timestamp time.Time `json:"timestamp"`
+
+	InputCount     uint8 `json:"inputcount"`
+	FCTOutputCount uint8 `json:"fctoutcount"`
+	ECOutputCount  uint8 `json:"ecoutcount"`
+}
+
+type FactoidTransactionIOs []FactoidTransactionIO
+
+type FactoidTransactionIO struct {
```

Comment:
I had a big thing written about how this should be done with a map. But I took some time to think it over and, hey, this works, and it preserves exact order of inputs and outputs in a simple way. It would be really nice if inputs and outputs had to be ordered but that's not the case so oh well. 

For users later wishing to construct transactions, catching duplicate addresses is important. But for fatd or pegnet, not so much. But the possibility of such should be taken into account. When parsing all amounts I still would recommend converting this to a map. but doing so for every transaction all the time would be more memory intensive.